File: src/main/scala/llmTests/a201603a/java/gpt35turbo/BagImpl.java
package llmTests.a201603a.java.gpt35turbo;

import java.util.*;

import llmTests.a201603a.java.Bag;
class BagImpl<X> implements Bag<X> {
    private final Map<X, Integer> bag;

    public BagImpl(Map<X, Integer> bag) {
        this.bag = bag;
    }

    public int numberOfCopies(X x) {
        return bag.getOrDefault(x, 0);
    }

    public int size() {
        int size = 0;
        for (int count : bag.values()) {
            size += count;
        }
        return size;
    }

    public List<X> toList() {
        List<X> list = new ArrayList<X>();
        for (Map.Entry<X, Integer> entry : bag.entrySet()) {
            X element = entry.getKey();
            int count = entry.getValue();
            for (int i = 0; i < count; i++) {
                list.add(element);
            }
        }
        return list;
    }
}

File: src/main/scala/llmTests/a201603a/scala/gpt35turbo/BagImpl.scala
//package llmTests.a201603a.scala.gpt35turbo
//
//import java.util
//import java.util.{List, Set}
//import java.util.function.Supplier
//import java.util.function.ToIntFunction
//import java.util.function.UnaryOperator
//
//import llmTests.a201603a.scala.Bag
//import llmTests.a201603a.scala.BagFactory
//import scala.jdk.CollectionConverters._
//
//class BagImpl[X](val map: Map[X, Int]) extends Bag[X] {
//  def numberOfCopies(x: X): Int = map.getOrElse(x, 0)
//
//  def size: Int = map.values.sum
//
//  def toList: util.List[X] = map.flatMap(kv => List.fill(kv._2)(kv._1)).toList.asJava
//}